sdk/python/kfp/dsl/_pipeline_param.py

```diff
@@ -12,14 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import re
-from collections import namedtuple
-from typing import List, Dict, Union
-
-# TODO: Move this to a separate class
-# For now, this identifies a condition with only "==" operator supported.
-ConditionOperator = namedtuple('ConditionOperator',
-                               'operator operand1 operand2')
-PipelineParamTuple = namedtuple('PipelineParamTuple', 'name op pattern')
+import attr
+from typing import List, Dict, Union, Any
+import warnings
+
+@attr.s
+class ConditionOperator:
```

Comment:
Should we name this `BinaryPredicate`?

Reply:
Sure! I just wanted to change as little as possible, but if you're ok with that --- I'd actually prefer that. Especially considering that `UnaryPredicate` would be useful too (e.g. for `__not__`).